Source/WebCore/page/Page.cpp

```diff
@@ -4110,6 +4111,21 @@ void Page::forEachDocument(const Function<void(Document&)>& functor) const
     forEachDocumentFromMainFrame(protectedMainFrame(), functor);
 }
 
+bool Page::findMatchingLocalDocument(const Function<bool(Document&)>& functor) const
+{
+    for (const auto* frame = &mainFrame(); frame; frame = frame->tree().traverseNext()) {
+        auto* localFrame = dynamicDowncast<LocalFrame>(frame);
```

Comment:
Should these use `RefPtr`?
```suggestion
    for (RefPtr frame = &mainFrame(); frame; frame = frame->tree().traverseNext()) {
        RefPtr localFrame = dynamicDowncast<LocalFrame>(frame);
```

Reply:
Oh yea, we should probably do that. Fixed.